### PR TITLE
Deploy other components of AAP

### PR DESCRIPTION
**Pull request by kdelee**
based on https://github.com/redhat-performance/ibm_cloud_tower_deploy/pull/30, figuring we will probably merge that first

This does a few things so far

1) Consolidate the several "add_.*_hosts" roles that added a set of hosts to a group into 1 role, that takes variables for the group name and hosts to be added
2) Move tasks into a few sets that can be executed with the `free` strategy because they have no interdependencies (e.g., create all the VPCs, create all the subnets, create all the ssh keys). Still can think about applying this to more steps in the process. I'm not 100% certain we can take full advantage of the free strategy if all tasks are executing on localhost. At worst, this is a reorganization that has no operational impact.
3) Generalizes a few actions to allow adding automationhub group and running the installer from it if there is no automation controller hosts indicated.
4) Updates inventory templates to include the automationhub group if hosts were deployed for it

### CODE REVIEW

**Comment by kdelee** [comment]
So far, I've run this locally with `scenario: automationhub-only` and it sucsessfully deployed the hosts, created the inventory, ran installer from the automationhub (because it was the only/first host in the control plane instances) and this installed hub/postgres on the db node.

File: vars/main.yml [hunk 1]
```diff
@@ -42,6 +42,6 @@ list_of_ports: [22, 80, 443, 5432, 9121, 9256, 9117, 9393, 9392, 9187, 9100, 271
 ssh_public_key: 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDlkniCZbiIhOv9RkoSE73Yg4sjRtuiuXWKHBCvhQ9BxZV33rMEGCUwYs2XT8QWbuJxBg+aXspZ1Of00lNV1S2/Ukg4n6Os7szI6mqJQgFR/oqLckGbHML76CdvwQ1iVoSFdE5bieQdfbK+LmIRVQAzO93090jWtWaJ8MkxzgarvXpPcwidVjB/PDsIq3NG/6boH/xUhxRH/Qt+4ShVH0/w4YTLGAvp56yeN+1XWoGy/D0V7JQX9y82ca3n9S+DsYAJBcUp/k7rjqtEsNBDYpNk7nvjmt9jiXCVCBu6EajcINOa/uEsWsyX/iyxE6XENsBP+hPK7/HMEqvMJw2cfJakCHCLTRpYZSOS4fursT53ZXt+UWko8S2xIqt/kkH4HSvrfHejlUWGILlRhEKX6SfMQaLLWaqz8QT9LIhEY/gRjk4wekAYWSQziaMeEcUsLVITihLKKG8qw6+ig5TOlw6LAhBJxCsKPkWLul7wAP5j6mI9/M73nX8JXX3O/drcNR0= pok@localhost.localdomain'
 ibmcloud_ssh_key: perfscale-automation-controller-ssh-key
 
-# Default values, override if you want to install tower or use a scenario as mapped in install_tower_vars.yml
+# Default values, override if you want to install tower or use a scenario as mapped in install_aap_vars.yml
```
Comment (jainnikhil30) [inline vars/main.yml:45]: should be deploy_aap_vars.yml ?

**Review by jainnikhil30 — APPROVED** [review]
looks good with that one small change in the main.yml comment.
Thanks Elijah

File: vars/main.yml (hunk 1 above)
Comment (jainnikhil30) [reply vars/main.yml:45]:
```suggestion
# Default values, override if you want to install tower or use a scenario as mapped in aap_deploy_vars.yml

```

**Pull request merged**